sotopia/envs/evaluators.py

```diff
@@ -168,8 +169,11 @@ class EnvResponsePlus(BaseModel):
 
 
 class EnvResponseGoalOnly(BaseModel):
```

Comment:
This can be a generic BaseModel

```python
EvauationDimensions = TypeVar("EvaluationDimensions", bound=BaseModel)

EvaluationForTwoAgents(BaseModel, Generic[EvaluationDimensions]):
    agent_1_evaluation: EvaluationDimensions
    agent_2_evaluation: EvaluationDimensions
```

Reply:
Then you can use e.g. `EvaluationForTwoAgents[EvaluationGoalOnly]` or anything